### ISSUE TITLE
Incompatibility with (RH)EL8

### ISSUE THREAD

**Issue by stejoo** [aisbergg/ansible-role-networkmanager#1]
Thanks for the role code :+1:
I have been using it for some time now, but ran into an issue today:

The role fails on a (current) RHEL8 system with the following error on task `networkmanager : create main NetworkManager configuration` (which deploys the `NetworkManager.conf` file):
```
AnsibleError: template error while templating string: no test named 'boolean'.
```
This happens because the template contains:
https://github.com/aisbergg/ansible-role-networkmanager/blob/bfc92b88219a69be97c24521abf64400ef93b49e/templates/NetworkManager.conf.j2#L6 

The `boolean` test was introduced in [Jinja 2.11.0](https://jinja.palletsprojects.com/en/2.11.x/changelog/#version-2-11-0). 
OK, good to know, but why am I creating this issue?
Well... your Galaxy metadata lists support for Enterprise Linux 8 (EL8): https://github.com/aisbergg/ansible-role-networkmanager/blob/49e049d8c710488740e8ca7dfcfec097fda42aaf/meta/main.yml#L15-L17
And I have recently started using your role in an environment that uses (up-to-date) Red Hat Enterprise Linux 8. RHEL8 currently ships with Jinja2 2.10.1 in their supported repositories. This causes my Ansible to bail out with the mentioned error...

I have been able to resolve the issue by more or less recreating the `boolean` test in the template. I have supplied my solution to this issue via pull request: <TODO>. Please have a look. Yes it's not as pretty as the `is boolean` test, but it does work on RHEL8 production systems with Ansible 2.9.27 and the shipped `python3-jinja2` from Red Hat for RHEL8.

**Comment by stejoo** [comment]
And in case decide against including my compatibility fix. Which I would absolutely understand because it's not as pretty as `boolean`. You could include something like this in a `requirements.txt` file.
```
# This role depends on PyPI components.
# To install the required version(s) on your system you can use a command as:
# `python -m pip --user install -r requirements.txt`
# See the pip requirements file documentation for details:
# https://pip.pypa.io/en/stable/user_guide/#requirements-files
#
# Tests run on the previous and current (latest) version of Ansible.
ansible>=2.9
# Some Jinja2 filters are used that are available in the newer releases.
jinja2>=2.11.0
```
(Modified from @robertdebock at https://github.com/robertdebock/ansible-role-tomcat/blob/7d6cb2f509384c23fab4a1249ffb875bdc28ec7d/requirements.txt )

Which specified the need for Ansible 2.9 and Jinja 2.11. :)

**Comment by aisbergg** [comment]
You are correct in the sense, that my listed version requirements are wrong. The wrong key is not `platforms` though, but `min_ansible_version`. In the current state it should contain `min_ansible_version: 2.11`. `platforms` usually refers to the target host OS platform, which the role can be applied to and not to the executing Ansible host (e.g. your development machine).

I see two options (besides your offered one):

- Bump `min_ansible_version` to 2.11 and call it a day. In that case you would have the following options:
    - You could install Ansible via PIP instead of DNF (if your policy allows it) to get the latest version and would then be able to use the role without modifications.
    - Include the missing test plugin in your Playbooks:
        1. Put the [test plugin](https://github.com/aisbergg/ansible-plugins/blob/master/test_plugins/boolean.py) into `./plugins/test/boolean.py` in the root of your Playbook dir
        2. Add following your custom plugins to your Playbooks by editing your `ansible.cfg`:
            ```ini
            # add local ./plugins/ path as source for plugins
            test_plugins       = ./plugins/test:~/.ansible/plugins/test:/usr/share/ansible/plugins/test
            filter_plugins     = ./plugins/filter:~/.ansible/plugins/filter:/usr/share/ansible/plugins/filter
            lookup_plugins     = ./plugins/lookup:~/.ansible/plugins/lookup:/usr/share/ansible/plugins/lookup
            library            = ./plugins/modules:~/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
            # ...
            ```

- Add compatibility with Ansible 2.10 by adding the boolean test directly to the role itself. I did this for some roles back then, but I dropped it eventually to reduce the code.

I would rather tend to the first option, even though that would mean worse backwards compatibility. What do think?

**Comment by stejoo** [comment]
You are absolutely right about the `platforms` part, my mistake there...

I forgot to look at the `min_ansible_version`. Upping that to `2.11` should not be necessary, because it's not Ansible that's going awry but Jinja2 is. 

If this were my role I would also prefer the first option over the second. Please spec the Jinja2 version requirement to clarify that one and be done with it. I can understand not supporting RHEL8 as Ansible control node from the point of view of reducing code base and making the test for boolean much cleaner.

Thank you for providing me with some tips on how to resolve the issue. Dropping the `boolean.py` into the `plugins/test/` directory in my Playbook directory sounds like a neat solution that might work for my client. Or I'll just keep a fork of this with my "fix" in their private repository, which is how it stands currently.

I think we can close this issue. :+1:
